crates/chainexec/src/output.rs

```diff
@@ -50,23 +50,24 @@ impl EpochExecutionOutput {
 
 /// Describes the output of executing a block.
 #[derive(Debug, Clone)]
-pub struct BlockExecutionOutput {
+pub struct BlockExecutionOutput<WB = WriteBatch, L = LogMessage> {
+    // TODO: Generic parameters, WB and L for transition
     /// State root as computed by the STF.
     computed_state_root: Buf32,
 
     /// Log messages emitted while executing the block.
     ///
     /// These will eventually be accumulated to be processed by ASM.
-    logs: Vec<LogMessage>,
+    logs: Vec<L>,
 
     /// Changes to the state we store in the database.
     ///
     /// This is NOT a state diff, that requires more precise tracking.
-    write_batch: WriteBatch,
+    write_batch: WB,
 }
 
-impl BlockExecutionOutput {
-    pub fn new(computed_state_root: Buf32, logs: Vec<LogMessage>, write_batch: WriteBatch) -> Self {
+impl<WB, L> BlockExecutionOutput<WB, L> {
+    pub fn new(computed_state_root: Buf32, logs: Vec<L>, write_batch: WB) -> Self {
```

Comment:
Why were these made generic?

Reply:
These are transitional. Since the current chainstate differs significantly from the OLState in the spec, I did not want to make intrusive changes on the `Chainstate` and hence a new `OLState` which should be merged/replaced in the future. The generic is just to support both, with default being the Chainstate. I guess Log generic can be removed since the `LogMessage` and `OLLog` are pretty similar and thus only one should exist.